### PR TITLE
Warn unused

### PR DESCRIPTION
**Pull request by Karsten1987**
fixes flakiness

### CODE REVIEW

File: rclcpp_lifecycle/test/test_state_wrapper.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
```
Comment (wjwwood) [inline rclcpp_lifecycle/test/test_state_wrapper.cpp]: 2017

File: rclcpp_lifecycle/test/test_state_wrapper.cpp [hunk 2]
```diff
@@ -40,11 +40,16 @@ TEST_F(TestStateWrapper, wrapper) {
...
   ASSERT_STREQ("my_c_state", c_state.label().c_str());
 
-  rcl_lifecycle_state_t * lc_state2 = new rcl_lifecycle_state_t {"my_c_state2", 3, NULL, NULL, 0};
-  rclcpp_lifecycle::State c_state2(lc_state2->id, lc_state2->label);
-  delete lc_state2;
-  lc_state2 = NULL;
-  EXPECT_EQ(3, c_state2.id());
-  EXPECT_FALSE(c_state2.label().empty());
-  ASSERT_STREQ("my_c_state2", c_state2.label().c_str());
+  // introduces flakiness
```
Comment (mikaelarguedas) [inline rclcpp_lifecycle/test/test_state_wrapper.cpp]: Is this ticketed somewhere ? or should we add a TODO here to remember that there is a commented out test?

**Comment by mikaelarguedas** [comment]
BTW: really like the description text of this PR :smile:

File: rclcpp_lifecycle/test/test_state_wrapper.cpp (hunk 2 above)
Comment (Karsten1987) [inline rclcpp_lifecycle/test/test_state_wrapper.cpp]: The commented out test is on purpose. When compiling with actual memory sanitizer, this last tests fails and thus produces a flaky test.

**Pull request merged**